client-ios/reducers/listReducer.js

```diff
@@ -22,15 +67,27 @@ export default (state = {
     case types.UPDATE_LISTS_STATE:
     return {
       ...state,
-      allListsDataSource: action.allListsDataSource,
-      allListsIsLoading: action.allListsIsLoading
+      dataSource: {
+        allListsDataSource: action.allListsDataSource,
+        singleListDataSource: []
+      },
+      ui: {
+        allListsIsLoading: action.allListsIsLoading,
+        singleListIsLoading: true
+      }
     };
 
     case types.UPDATE_SINGLE_LIST_STATE:
     return {
       ...state,
-      singleListDataSource: action.singleListDataSource,
-      singleListIsLoading: action.singleListIsLoading
+      dataSource: {
+        singleListDataSource: action.singleListDataSource,
+        allListsIsLoading: []
```

Comment:
and....does UPDATE_SINGLE_LIST_STATE action always reset allListsIsLoading(I think this is typo) with empty array? This means that when we update single list, we lost all lists state from the store